### PR TITLE
[david] add support for path

### LINKED ISSUE #637: Support David's path query parameter
Hi,

David [now supports](http://davidiswatching.tumblr.com/post/132882415144/non-root-packagejson) `package.json` files which aren't in the root of the GitHub repository through a `path` query parameter.

That's how

``` md
![devDependency Status](https://david-dm.org/algolia/algoliasearch-zendesk/dev-status.svg?path=app)
```

renders like this: [devDependency Status](https://david-dm.org/algolia/algoliasearch-zendesk?path=app#info=devDependencies) , based on the `package.json` in the `app/` directory of this project instead of the root. I'd like to see shield.io also support it.

I'm not really sure of how it could be handled at the URL level, maybe just passing along the query parameter? Or maybe just adding it after the repo name?

``` txt
https://img.shields.io/david/<type>/<repo owner>/<repo name><path to the package.json>.svg
https://img.shields.io/david/dev/algolia/algoliasearch-zendesk/app.svg
```

Would that make sense for you?

### PR DESCRIPTION
**Pull request by rraziel**
This PR adds support for David's `path` query parameter (used to specify in which folder the package.json file is located).

The diff is a bit messy because the indentation had to be changed, but all it really does is adding the path query param declaration and an if block to append `?path=x` when sending the HTTP request.

Note: this closes #637

### CODE REVIEW

File: services/david/david.tester.js [hunk 1]
```diff
@@ -37,6 +37,13 @@ t.create('david peer dependencies (valid)')
     value: isDependencyStatus
   }));
 
+t.create('david dependencies (path)')
+  .get('/babel/babel.json?path=packages/babel-core')
+  .expectJSONTypes(Joi.object().keys({
+    name: 'dependencies',
+    value: isDependencyStatus
+  }));
+
```
Comment (RedSparr0w) [inline services/david/david.tester.js]: Could you also add a test for when `?path` is an invalid path.

**Comment by RedSparr0w** [comment]
Side note you can see a cleaner diff by hiding the whitespace changes:
image

File: services/david/david.tester.js (hunk 1 above)
Comment (rraziel) [reply services/david/david.tester.js]: Will do.

**Comment by RedSparr0w** [comment]
Thanks for the changes, just noticed 1 more thing.
Could you update the title to include `[david]` as this allows the service test to be run on this PR.
eg: `[david] add support for path`

**Comment by rraziel** [comment]
Hm just to make sure: you meant change the PR title or the commit message?

**Comment by rraziel** [comment]
Ok then the way I changed it should be fine (I think?)

**Comment by RedSparr0w** [comment]
Yeah looks good, I think closing and reopening the PR will re run the test.

**Comment by RedSparr0w** [comment]
Would you be able to make some sort of minor change to the path not found test or something please?
Seems it needs another commit for the `services-PR` tests to run correctly.

**Comment by RedSparr0w** [comment]
Was failing due to something unrelated to this PR, have rerun test and passing now.

Approved!
👍

**Pull request merged**